pkg/restic/config.go

```diff
@@ -0,0 +1,70 @@
+package restic
+
+import (
+	v1beta1_api "github.com/appscode/stash/apis/stash/v1beta1"
+	shell "github.com/codeskyblue/go-sh"
+)
+
+type ResticWrapper struct {
+	sh     *shell.Session
+	config SetupOptions
+}
+
+type BackupOptions struct {
+	BackupDirs []string
+	Cleanup    CleanupOptions
+}
+
+type RestoreOptions struct {
+	Rules []v1beta1_api.Rule
```

Comment:
Remove API dependency

```
type RestoreOptions struct {
	Host        string // use this instead of SetupOptions.Hostname
	RestoreDirs []string
	Snapshots   []string // when Snapshots are specified Host and RestoreDirs will not be used
}
```

Use a helper to get `RestoreOptions` from `api.Rules`

```
// return first matching rule
// if hosts is empty for any rule, it will match any hostname
func RestoreOptionsForHost(hostname string, rules []v1beta1.Rule) RestoreOptions {
	for _, rule := range rules {
		if len(rule.Hosts) == 0 {
			return RestoreOptions{
				Host:        "",
				RestoreDirs: rule.Paths,
				Snapshots:   rule.Snapshots,
			}
		}
		if strings.Contains(rule.Hosts, hostname) {
			return RestoreOptions{
				Host:        hostname,
				RestoreDirs: rule.Paths,
				Snapshots:   rule.Snapshots,
			}
		}
	}
	return RestoreOptions{}
}
```